Eatery/Controllers/BRB Feature/BRBConnectionHandler.swift

```diff
@@ -277,14 +128,14 @@ class BRBConnectionHandler: WKWebView, WKNavigationDelegate {
         getHTML(block: { (html: NSString) -> () in
             if self.failedToLogin() {
                 self.stage = .loginFailed
-            } else if self.url!.absoluteString.contains(updateProfileURLString) {
-                self.stage = .loginFailed
             } else if html.contains("<h1>CUWebLogin</h1>") {
                 self.stage = .loginScreen
-            } else if self.url!.absoluteString == fundsHomeURLString {
-                self.stage = .fundsHome
-            } else if self.url!.absoluteString == diningHistoryURLString {
-                self.stage = .diningHistory
+            } else if self.url?.absoluteString.contains("sessionId") ?? false {
+                guard let url = self.url, let urlComponents = URLComponents(url: url, resolvingAgainstBaseURL: true) else { return }
+
+                if let sessionId = urlComponents.queryItems?.first(where: { $0.name == "sessionId" })?.value {
```

Comment:
If the session id is not available, this seems like it will keep the app stuck on loginScreen forever. Is this intended?